### DISCUSSION TITLE
TimePickerType and EventListener explanation of begin time field

### DISCUSSION THREAD

**Discussion by lnNgyn1** [kimai/kimai#4605] [General] [answered]
Hi Kevin, I'm working on adding the begin_time field to the period insert plugin, and I'm struggling to figure out how to take the given input (7:00 AM for example) and add that to the new timesheet entries. My understanding is that it returns a string, and I have to convert it into a `DateTime` object and call `SetBegin()` on a `Timesheet` object, but that hasn't been working for me. Is there a piece of code somewhere where this process occurs? I'm also trying to understand the event listeners and mapping the begin field to the end and duration fields. The latter concern isn't as important.

For reference, I'm looking at [this function](https://github.com/kimai/kimai/blob/df3ca9d5a9176d81c5adfd5dace1c0e3218bd26b/src/Form/TimesheetEditForm.php#L165) in the `TimesheetEditForm.php` file.

**Comment by kevinpapst** [comment, marked as answer]
It takes a DateTime and gives a DateTime (not sure if DateTimeInterface/DateTimeImmutable is yet supported, use that if possible)

**Reply by lnNgyn1** [reply]
Okay, thanks! Removing a naming error somewhere else and removing the mapped key fixed it.

**Reply by lnNgyn1** [reply]
I hope it's okay if I ask in this thread, but another question I have is the YesNoType checkbox - how can I set the checkbox as checked out by default? Right now I'm taking a look at the mark as exported option on the export page.

**Reply by kevinpapst** [reply]
"what is checked out"?
Anyway: you know that these fields reflect the properties of your entity, right?
So a checkbox for the field exported is usually not checked (exported = false) until the timesheet is really exported. 
Then the checkbox is checked (exported = true).
